### ISSUE TITLE
Tap tempo

### ISSUE THREAD

**Issue by danboid** [muse-sequencer/muse#392]
This is definitely a "nice to have"-style request but it would be cool to be able to set the tempo with a couple of clicks on what is currently just the Tempo label.

**Comment by danvd** [comment]
Hi Dan,

Made this. Can you check?

**Comment by danboid** [comment]
Unfortunately this isn't working for me. I've tried it at least 20/30+ times and only once did the BPM change after clicking twice on the TAP button but I'm not sure if the BPM was (roughly) correct that one time it kinda worked.

**Comment by danvd** [comment]
It works like this:
First 3 times it only collects data (algorithm uses average data), and only after 3rd click changes tempo value. When button is not clicked during 2 seconds, the tapping state is resetted.

Is this behaves right? Honestly I did not used tap tempo before. Correct me, if it should work differently.

**Comment by danvd** [comment]
You can give me an example of program to test with.

**Comment by danboid** [comment]
Most tempo tap features I've seen in the past only require two clicks to set a new tempo but I can understand why having it require 3 or 4 may produce more accurate results.

I've tried three clicks but it still doesn't work most of the time.

**Comment by danboid** [comment]
I think REAPER is one DAW that implements this.

**Comment by danboid** [comment]
Tracktion has a tempo-tap feature too if you want to try a Linux-native DAW.

**Comment by danboid** [comment]
If you have any Android devices you might find it easier to try the tempo tap feature in PracticeHub which is available from F-droid.

**Comment by danvd** [comment]
Yes, I have lots of them. Thanks for the idea :)

**Comment by danvd** [comment]
Ok. Tried it. Will do like that now.

**Comment by danvd** [comment]
Pushed a fix. Can you retest again?

**Comment by danboid** [comment]
Seems to work well enough now!

Thanks Andrew!
